fiftyone/migrations/revisions/v0_18_1.py

```diff
@@ -0,0 +1,147 @@
+"""
+FiftyOne v0.18.1 revision.
+
+| Copyright 2017-2022, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+from datetime import datetime
+import logging
+import string
+
+
+logger = logging.getLogger(__name__)
+
+
+def up(db, dataset_name):
+    match_d = {"name": dataset_name}
+    dataset_dict = db.datasets.find_one(match_d)
+
+    if "url_name" not in dataset_dict:
+        _set_url_name(db, dataset_dict)
+
+    if "description" not in dataset_dict:
+        dataset_dict["description"] = None
+
+    db.datasets.replace_one(match_d, dataset_dict)
+
+
+def down(db, dataset_name):
+    match_d = {"name": dataset_name}
+    dataset_dict = db.datasets.find_one(match_d)
+
+    dataset_dict.pop("url_name", None)
+    dataset_dict.pop("description", None)
+
+    db.datasets.replace_one(match_d, dataset_dict)
+
+
+def _set_url_name(db, dataset_dict):
+    _id = dataset_dict["_id"]
+    name = dataset_dict.get("name", None)
+    url_name = dataset_dict.get("url_name", None)
+
+    existing_names = set()
+    existing_url_names = set()
+
+    for d in db.datasets.find(
+        {"_id": {"$ne": _id}}, {"name": 1, "url_name": 1}
+    ):
+        if "name" in d:
+            existing_names.add(d["name"])
+
+        if "url_name" in d:
+            existing_url_names.add(d["url_name"])
+
+    if name is None:
```

Comment:
could name ever be none? i thought name is required for a dataset

Reply:
It shouldn't be possible, but I like to be extra extra careful in migrations